### PR TITLE
mlh: update Jenkins jobs following removal of kernel 4.9 support

### PR DESCRIPTION
**Pull request by nbusseneau**
We removed support for kernel 4.9 in the CI via
bbda112037b8d95f8c084b81dd1dc2d0bc4aff82, as part of ongoing work to bump the minimal supprted kernel version (see [1]).

We have rotated / expanded the Jenkins test jobs as follow:

- Changed: K8s 1.16 on Kernel 4.19 (instead of 4.9, triggered on `/test`).
- Changed: K8s 1.17-1.24 on Kernel 4.19 (instead of 4.9, triggered on `/test-missed-k8s`).

See the Table of Truth™️ for up to date status on all trigger phrases: https://docs.google.com/spreadsheets/d/1TThkqvVZxaqLR-Ela4ZrcJ0lrTJByCqrbdCjnI32_X0/edit#gid=0

[1]: https://github.com/cilium/cilium/issues/22116

### CODE REVIEW

**Comment by nbusseneau** [comment]
@aanm @joestringer The following PR checks need to be marked as required in GH settings for `master` branches:

- `k8s-1.16-kernel-4.19 (test-1.16-4.19)`

The following need to be unmarked:

- `k8s-1.16-kernel-4.9 (test-1.16-4.9)`

**Comment by nbusseneau** [comment]
We don't need to wait for any CI run to complete, I triggered the CI only to verify trigger phrases and also have GH pick up the new jobs (so that we can mark them as required), but this change is completely independent of CI and only requires a reviewer's approval.

File: .github/maintainers-little-helper.yaml
Comment (pchaigno) [inline .github/maintainers-little-helper.yaml:117]: It seems these lists all have a new entry now for 1.25. Why?
I think I'm missing how those `correlate-with-stable-jobs` are used.

Comment (nbusseneau) [reply .github/maintainers-little-helper.yaml:117]: MLH correlation uses the matrix defined in config to correlate failures in PR jobs with failures in scheduled jobs on `master`.

For example, related to the lines outlined above: `Cilium-PR-K8s-1.19-kernel-4.19` (i.e. `
k8s-1.19-kernel-4.19 (test-1.19-4.19)`) will be correlated with all 4.19 `master` jobs.

The reason 1.25 is a "new" entry is because we already had 4.19 testing on 1.25, so now that 1.16-1.24 also use 4.19 it makes sense to correlate against it as well since it's the same kernel.

Comment (pchaigno) [reply .github/maintainers-little-helper.yaml:117]: Got it. Thanks for the explanation!

**Review by pchaigno — APPROVED** [review]

**Pull request merged**
